### PR TITLE
437334 sonarcloud updates

### PR DESCRIPTION
**Pull request by feedmypixel**
SonarCloud updates to the Frontend template 

Screenshot 2024-09-16 at 09 43 07

### CODE REVIEW

File: .github/workflows/publish-hotfix.yml [hunk 1]
```diff
@@ -22,14 +22,16 @@ jobs:
...
           cache: npm
-
-      # Ensure tests pass. Add more checks if required
-      - run: npm ci
-      - run: npm test
+        run: |
+          npm ci
+          npm run format:check
+          npm run lint
+          npm test
```
Comment (flurdy) [inline .github/workflows/publish-hotfix.yml:34]: Does the step still fail if e.g the lint fails but the test passes?

Comment (feedmypixel) [reply .github/workflows/publish-hotfix.yml:34]: If the lint fails then this step fails. It won't get to `npm test`

Comment (flurdy) [reply .github/workflows/publish-hotfix.yml:34]: Ok just wondered as it was wrapped as a one command step

File: src/config/nunjucks/filters/format-date.test.js [hunk 2]
```diff
@@ -13,20 +13,24 @@ describe('#formatDate', () => {
...
   describe('With format attribute', () => {
     test('Date should be in provided format', () => {
       expect(
         formatDate(
-          '2022-01-17T11:40:02.242Z',
+          '2023-04-08T11:40:02.242Z',
           "h:mm aaa 'on' EEEE do MMMM yyyy"
         )
-      ).toBe('11:40 am on Monday 17th January 2022')
+      ).toBe('12:40 pm on Saturday 8th April 2023')
```
Comment (flurdy) [inline src/config/nunjucks/filters/format-date.test.js:33]: Why 12:40?

File: src/server/common/helpers/serve-static-files.js [hunk 3]
```diff
@@ -18,8 +19,8 @@ export const serveStaticFiles = {
           },
           method: 'GET',
           path: '/favicon.ico',
-          handler(request, h) {
-            return h.response().code(204).type('image/x-icon')
+          handler(_request, h) {
```
Comment (flurdy) [inline src/server/common/helpers/serve-static-files.js:22]: I assume the underscore is some node convention?

Comment (feedmypixel) [reply src/server/common/helpers/serve-static-files.js:22]: Not `Node.js` per say. More JS/TS convention for unused args. One that SonarCloud picks up on

**Review by flurdy — APPROVED** [review]

**Pull request merged**